libcudacxx/test/libcudacxx/cuda/try_cancel/try_cancel.pass.cpp

```diff
@@ -0,0 +1,220 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: pre-sm-100
+// UNSUPPORTED: !nvcc
+// UNSUPPORTED: nvrtc
+// UNSUPPORTED: c++98, c++03, c++14, c++17
+
+#include <cuda/try_cancel>
+
+__device__ void vec_add_impl1(int* a, int* b, int* c, int n, dim3 tb)
+{
+  int idx = threadIdx.x + tb.x * blockDim.x;
+  if (idx < n)
+  {
+    c[idx] += a[idx] + b[idx];
+  }
+}
+
+__device__ void vec_add_impl2(int* a, int* b, int* c, int n, dim3 tb)
+{
+  int x_idx = threadIdx.x + (tb.x * blockDim.x);
+  int y_idx = threadIdx.y + (tb.y * blockDim.y);
+  int idx   = x_idx + y_idx * (blockDim.x * gridDim.x);
+  if (idx < n)
+  {
+    c[idx] += a[idx] + b[idx];
+  }
+}
+
+__device__ void vec_add_impl3(int* a, int* b, int* c, int n, dim3 tb)
+{
+  int x_idx = threadIdx.x + (tb.x * blockDim.x);
+  int y_idx = threadIdx.y + (tb.y * blockDim.y);
+  int z_idx = threadIdx.z + (tb.z * blockDim.z);
+  int idx   = x_idx + y_idx * (blockDim.x * gridDim.x) + z_idx * (blockDim.x * gridDim.x * blockDim.y * gridDim.y);
+  if (idx < n)
+  {
+    c[idx] += a[idx] + b[idx];
+  }
+}
+
+__global__ void vec_add_det1(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::__detail::__for_each_cancelled_block<1>(threadIdx.x == tidx, [=](dim3 tb) {
+    vec_add_impl1(a, b, c, n, tb);
+  });
+}
+
+__global__ void vec_add_det2(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::__detail::__for_each_cancelled_block<2>(threadIdx.x == tidx && threadIdx.y == tidx, [=](dim3 tb) {
+    vec_add_impl2(a, b, c, n, tb);
+  });
+}
+
+__global__ void vec_add_det3(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::__detail::__for_each_cancelled_block<3>(
+    threadIdx.x == tidx && threadIdx.y == tidx && threadIdx.z == tidx, [=](dim3 tb) {
+      vec_add_impl3(a, b, c, n, tb);
+    });
+}
+
+__global__ void vec_add1(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::for_each_cancelled_block<1>([=](dim3 tb) {
+    vec_add_impl1(a, b, c, n, tb);
+  });
+}
+
+__global__ void vec_add2(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::for_each_cancelled_block<2>([=](dim3 tb) {
+    vec_add_impl2(a, b, c, n, tb);
+  });
+}
+
+__global__ void vec_add3(int* a, int* b, int* c, int n, int tidx = 0)
+{
+  cuda::experimental::for_each_cancelled_block<3>([=](dim3 tb) {
+    vec_add_impl3(a, b, c, n, tb);
+  });
+}
+
+#include <iostream>
+#include <vector>
```

Comment:
Please put includes at the top of the file.

Reply:
This helps test that the API is used exclusively with its include file. If we put these two at the top of the file it could be that the API can only be used if these two headers are also included when using it.

Reply:
Alright, fine!